### PR TITLE
Map read-only collections by default

### PR DESCRIPTION
**Pull request by lbargaoanu**

### CODE REVIEW

File: src/AutoMapper/TypeDetails.cs [hunk 1]
```diff
@@ -125,40 +123,28 @@ where extensionMethod.IsGenericMethodDefinition
...
-            var filteredMembers = allMembers
+            allMembers
                 .OfType<PropertyInfo>()
                 .GroupBy(x => x.Name) // group properties of the same name together
-                .Select(x =>
-                    x.Any(y => y.CanWrite && y.CanRead)
-                        ? // favor the first property that can both read & write - otherwise pick the first one
-                        x.First(y => y.CanWrite && y.CanRead)
-                        : x.First())
-                .Where(pi => pi.CanWrite || pi.PropertyType.IsListOrDictionaryType())
```
Comment (lbargaoanu) [inline src/AutoMapper/TypeDetails.cs:156]: I changed this so we try to map now _all_ get-only collections. If you think is too impactful, I guess we can continue to say "backward compatibility" and keep the check.

**Comment by jbogard** [comment]
Merge conflict on the api compat file

**Comment by lbargaoanu** [comment]
@jbogard Can we merge this?

**Comment by lbargaoanu** [comment]
@jbogard Ping :)

**Comment by lbargaoanu** [comment]
@jbogard Ping :)

**Comment by jbogard** [comment]
More merge conflicts :(

**Comment by lbargaoanu** [comment]
Rebased. I'll check too :)

**Comment by lbargaoanu** [comment]
I say let's ignore ApiCompat and check it at the end when we merge all the PR-s. Once we merge this one, I'll rebase the others.

**Comment by lbargaoanu** [comment]
This one and #3379 are good to go.

**Pull request merged**